src/compile/axis.ts

```diff
@@ -177,22 +178,32 @@ export namespace properties {
       }, labelsSpec);
     }
 
+    let textTemplate = '{{ datum.data }}';
+
     if (contains([NOMINAL, ORDINAL], fieldDef.type) && axis.labelMaxLength) {
       // TODO replace this with Vega's labelMaxLength once it is introduced
-      labelsSpec = extend({
-        text: {
-          template: '{{ datum.data | truncate:' + axis.labelMaxLength + '}}'
-        }
-      }, labelsSpec || {});
+      textTemplate = '{{ datum.data | truncate:' + axis.labelMaxLength + '}}';
```

Comment:
I mean, if you really can't find where .. ask me.  But it's good to practice venturing into other codebase such as Vega. :)